postgres/__init__.py

```diff
@@ -838,6 +840,30 @@ def make(self, values):
             instance = ModelSubclass(record)
             return instance
 
+        def parse(self, s, curs, retry=False):
+            # Override to protect against race conditions:
+            #   https://github.com/gratipay/postgres.py/issues/26
+
+            if s is None:
+                return None
+
+            tokens = self.tokenize(s)
+            if len(tokens) != len(self.atttypes):
+                # The number of columns has changed, re-fetch the type info
+                self.__dict__.update(self._from_db(self.name, curs).__dict__)
```

Comment:
I see, we're calling `_from_db`, which is a constructor classmethod, and then we're overwriting all possible attributes on ourself with the corresponding attributes from the new instance.